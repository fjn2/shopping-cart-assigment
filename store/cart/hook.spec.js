import { mount } from 'enzyme';
import { useEffect } from 'react';
import useCart, { CartContext } from './hook';

describe('Cart hook', () => {
  it('Should initialize the cart correctly', () => {
    // given
    const TestConsumer = () => {
      const { cart } = useCart()
      return JSON.stringify(cart)
    }

    // when
    const elem = mount(<CartContext><TestConsumer /></CartContext>)
    
    // then
    expect(elem).toMatchSnapshot()
  })

  it('Should add items to the cart', () => {
    // given
    const testProduct = {
      id: 1,
      name: 'Test Name',
      price: 123,
      brand: 'Test brand',
      quantity: 2,
      itemOptions: [{ color: ['red'] }]
    }

    // when
    const TestConsumer = () => {
      const { addItem, cart } = useCart()
      useEffect(() => {
        addItem(testProduct)
      }, [])
      return JSON.stringify(cart)
    }
    const elem = mount(<CartContext><TestConsumer /></CartContext>)

    // then
    expect(elem).toMatchSnapshot()
  })

  it('Should remove an item from the cart correctly', () => {
    // given
    const testProduct = {
      id: 1,
      name: 'Test Name',
      price: 123,
      brand: 'Test brand',
      quantity: 2,
      itemOptions: [{ color: ['red'] }]
    }
    const TestConsumer = () => {
      const { cart, addItem, removeItem } = useCart()
      useEffect(() => {
        addItem(testProduct)
        removeItem(Object.keys(cart)[0])
      }, [])
      return JSON.stringify(cart)
    }

    const elem = mount(<CartContext><TestConsumer /></CartContext>)
    
    expect(elem).toMatchSnapshot()
  })

  it('Should update an item from the cart correctly', () => {
    // given
    const testProduct = {
      id: 1,
      name: 'Test Name',
      price: 123,
      brand: 'Test brand',
      quantity: 2,
      itemOptions: [{ color: ['red'] }]
    }
    const updatedProduct = {
      id: 1,
      name: 'Test Name',
      price: 123,
      brand: 'Test brand',
      quantity: 5, // <- new quantity
      itemOptions: [{ color: ['red'] }]
    }
    const TestConsumer = () => {
      const { cart, addItem, updateItem } = useCart()
      useEffect(() => {
        addItem(testProduct)
        updateItem(updatedProduct)
      }, [])
      return JSON.stringify(cart)
    }

    const elem = mount(<CartContext><TestConsumer /></CartContext>)
    
    expect(elem).toMatchSnapshot()
  })
})