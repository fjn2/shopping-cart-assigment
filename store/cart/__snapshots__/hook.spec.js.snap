// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart hook Should add items to the cart 1`] = `
<CartContext>
  <TestConsumer>
    {"{\\"id\\":1,\\"name\\":\\"Test Name\\",\\"price\\":123,\\"itemOptions\\":[{\\"color\\":[\\"red\\"]}]}":{"id":1,"name":"Test Name","price":123,"brand":"Test brand","quantity":2,"itemOptions":[{"color":["red"]}]}}
  </TestConsumer>
</CartContext>
`;

exports[`Cart hook Should initialize the cart correctly 1`] = `
<CartContext>
  <TestConsumer>
    {}
  </TestConsumer>
</CartContext>
`;

exports[`Cart hook Should remove an item from the cart correctly 1`] = `
<CartContext>
  <TestConsumer>
    {}
  </TestConsumer>
</CartContext>
`;

exports[`Cart hook Should update an item from the cart correctly 1`] = `
<CartContext>
  <TestConsumer>
    {"{\\"id\\":1,\\"name\\":\\"Test Name\\",\\"price\\":123,\\"itemOptions\\":[{\\"color\\":[\\"red\\"]}]}":{"id":1,"name":"Test Name","price":123,"brand":"Test brand","quantity":5,"itemOptions":[{"color":["red"]}]}}
  </TestConsumer>
</CartContext>
`;
